helmclient.go

```diff
@@ -299,7 +299,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 		if err != nil {
 			return microerror.Mask(err)
 		}
-	} else {
+	} else if installTiller && upgradeTiller {
```

Comment:
Currently this error is returned if both installTiller and upgradeTiller are false. But this is a valid case. 

installTiller is false because the deployment has already been created but the pod is not ready yet.
We should continue and wait for the ready pod which is the next step. 

Now we just error if both flags are true. This should not be possible but I think its good to error as it means something is very wrong.